instrumentation/spring/spring-integration-4.1/javaagent/src/test/groovy/SpringIntegrationAndRabbitTest.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+
+class SpringIntegrationAndRabbitTest extends AgentInstrumentationSpecification implements WithRabbitProducerConsumerTrait {
+  def setupSpec() {
+    startRabbit()
+  }
+
+  def cleanupSpec() {
+    stopRabbit()
+  }
+
+  def "should cooperate with existing RabbitMQ instrumentation"() {
+    when:
+    producerContext.getBean("producer", Runnable).run()
+
+    then:
+    assertTraces(2) {
+      trace(0, 7) {
+        span(0) {
+          name "producer"
+        }
+        span(1) {
+          name "testProducer.output"
+          childOf span(0)
+        }
+        span(2) {
+          name "exchange.declare"
+          childOf span(1)
+          kind CLIENT
+        }
+        span(3) {
+          name "testTopic -> testTopic send"
+          childOf span(1)
+          kind PRODUCER
+        }
+        span(4) {
+          name ~/testTopic.anonymous.[-\w]+ process/
+          childOf span(3)
+          kind CONSUMER
+        }
+        span(5) {
+          name "testConsumer.input"
+          childOf span(3)
+        }
```

Comment:
This test runs both RabbitMQ and spring-integration instrumentation - the interesting thing here is that the "consumer" span created by spring-integration instrumentation is the child of the actual PRODUCER span (and sibling of CONSUMER) - that's another consequence of choosing the context extracted from the message.

Removing the integration span would be even worse here, since now the parent of whatever comes next would be the PRODUCER span, not the CONSUMER one.

Reply:
Which of the spans above is created by spring-integration? 4 or 5?

Reply:
Number 5; number 4 (the CONSUMER one) is created by rabbit instrumentation.

Reply:
I don't like it... I am not sure I want to block this PR, but at some time in near future we have to review span modelling here...

Reply:
I have mixed feeling about it too - this situation was the reason why my first PR had initially used links for spans extracted from incoming messages.

There's one more idea that comes to my mind, a bit more complex behavior:
* if there's no "local" current span, use the span extracted from the message as the parent;
* otherwise use the "local" span as the parent and link the span context extracted from the message (if it's a different span than parent)

Reply:
We can discuss this tomorrow face to face, if you wish

Reply:
Yep, definitely! 👍